pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -415,3 +420,15 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+    @staticmethod
+    def _clean_logger(trainer: "pl.Trainer") -> None:
+        loggers = trainer.logger._logger_iterable if isinstance(trainer.logger, LoggerCollection) else [trainer.logger]
+        for logger in loggers:
+            if isinstance(logger, TensorBoardLogger) and logger._experiment is not None:
+                rank_zero_warn(
+                    "When using `ddp_spawn`, the `TensorBoardLogger` experiment should be `None`. Setting it to `None`."
+                )
+                del logger._experiment
+                logger._experiment = None
+                gc.collect()
```

Comment:
adding collect here is overkill. and also a source of slowness

if the reference count for the object in `logger._experiment` is 0 after deletion, python will instantly `__del__` it.

```suggestion
```

And if the reference count is >0, neither `del` nor `gc.collect()` will help

Reply:
See answer `2)` here: https://stackoverflow.com/a/6772907